pydicom/dataset.py

```diff
@@ -1992,6 +1989,39 @@ def from_json(cls, json_dataset, bulk_data_uri_handler=None,
             dataset.add(data_element)
         return dataset
 
+    def to_json_dict(self, bulk_data_threshold=1,
+                     bulk_data_element_handler=None):
+        """Return a dictionary representation of the :class:`Dataset`
+        conforming to the DICOM JSON Model as described in the DICOM
+        Standard, Part 18, :dcm:`Annex F<part18/chaptr_F.html>`.
+
+        Parameters
+        ----------
+        bulk_data_threshold : int, optional
+            Threshold for the length of a base64-encoded binary data element
+            above which the element should be considered bulk data and the
+            value provided as a URI rather than included inline (default:
+            ``1``).
+        bulk_data_element_handler : callable, optional
+            Callable function that accepts a bulk data element and returns a
+            JSON representation of the data element (dictionary including the
+            "vr" key and either the "InlineBinary" or the "BulkDataURI" key).
+
+        Returns
+        -------
+        dict
+            :class:`Dataset` representation based on the DICOM JSON Model.
+        """
+        json_dataset = {}
+        for key in self.keys():
+            json_key = '{0:04x}{1:04x}'.format(key.group, key.element).upper()
```

Comment:
You can use X for upper case hex: `'{:04X}{:04X}'.format(key.group, key.element)` or perhaps just `{:08X}.format(key)` although that form is less obvious its a tag...

Reply:
Yes, I just copied that... good point.